Manifest update and URL-based downloads

deepayan opened the pull request:
`nhanesManifest()` can now get both public and limited access files. Also added variants of `nhanes()` and `nhanesCodebook()` to work off URLs (from the manifest) rather than table names.

cjendres1 commented on R/nhanes.R:
Questions/comments regarding nhanesFromURL
1: Is there any instance where this `nh_table <- file_path_sans_ext(basename(url))` would not work?
If that pulls nh_table then perhaps nh_table should be removed as an argument to avoid unintentional inconsistency.
2: It appears that nh_table is only needed when translated=TRUE. Recommend moving `guess table name` into the `if translated` section.
3: Prefix should probably be defined in nhanes_constants.R. For example: cdcURL <- 'https://wwwn.cdc.gov/'
In that case, prefix would also be removed from the argument list.

deepayan replied:
1. You are right, I can't think of any current use case where that would be useful. We can omit it now, and add it back later if we need to.
2. Yes, we should do that.
3. The idea was to give the user the option to change this to point to local copies of files, which would not be possible if the prefix was a constant. However, we are now using the env var `NHANES_TABLE_BASE` to choose an alternative source during startup. We could use that and define
```r
prefix <- Sys.getenv("NHANES_TABLE_BASE", unset = "https://wwwn.cdc.gov")
```
inside the function instead of setting it as an argument.

I'll discuss with Robert and update the PR tomorrow.

deepayan commented:
`nhanesFromURL()` etc. are now simplified, with fewer arguments. I implemented the prefix 'constants' using active bindings, so that they can be changed dynamically using an environment variable.
